config/routes.rb

```diff
@@ -145,6 +145,9 @@
       resources :review_immunity_details, only: %i[edit update show]
 
       resources :permitted_development_rights, only: %i[new create edit update show]
+      get "/cil_liability/edit", action: :edit, controller: "cil_liability"
+      post "/cil_liability", action: :update, controller: "cil_liability"
+      patch "/cil_liability", action: :update, controller: "cil_liability"
```

Comment:
Can we not use the regular `resources` style of routing here

Perhaps need to do `resource` if the edit/update actions aren't actually modifying a new instance of cil_liability

Reply:
I couldn't figure out the syntax for `resources` or `resource` — rails seems to be trying to be too clever. (Perhaps because `cil_liability` isn't a model, just a property of `PlanningApplication`. Rails wanted to add an ID to the url for the particular instance of `cil_liability`, which of course doesn't exist.)

Reply:
Yeah I think that's why I added before something like `resource :legislation, only: %i[show update]` because legislation isn't actually a model and using resources was trying to add a legislation_id in the routing

Reply:
Maybe you've already seen but this gives a quick breakdown - https://www.rubyinrails.com/2019/04/16/rails-routes-difference-between-resource-and-resources/

Reply:
>Moreover, it still directs requests to pluralized controller name.

Ah this was part of the issue I was having: it was pluralising even when I didn't want it to. But maybe I can use `resource` and only override the controller name.